library/src/tensor_operation_instance/gpu/mha/CMakeLists.txt

```diff
@@ -39,7 +39,12 @@ else()
   file(STRINGS ${FMHA_CPP_FOLDER}/blob_list.txt FMHA_FWD_GEN_BLOBS)
 endif()
 
-# actually generate the cpp files
+
+# actually generate the cpp files but before that delete the blob file, if it exists to avoid append of old content.
+if(EXISTS ${FMHA_CPP_FOLDER}/blob_list.txt)
+    file(REMOVE ${FMHA_CPP_FOLDER}/blob_list.txt)
+endif()
+
```

Comment:
I think we should be checking / removing the blob_list.txt file above where we output the original blob_list.txt (lines 31-40). I think that way it'll be freshly generated each time, and not removed after generating.